webdriver-extension/index.bs

```diff
@@ -0,0 +1,546 @@
+<dl>
+  <dt><a>Mock Sensor Reading Values</a></dt>
+  <dd>
+    <pre class="idl">
+      dictionary GeolocationReadingValues {
+        required double? latitude;
+        required double? longitude;
+        required double? altitude;
+        required double? accuracy;
+        required double? altitudeAccuracy;
+        required double? heading;
+        required double? speed;
+      };
+    </pre>
+  </dd>
+</dl>
+
+### Proximity Sensor ### {#mock-proximity-sensor}
+The <dfn for="MockSensorType" enum-value>"proximity"</dfn> type is the mock sensor type
+associated with the usage of the <a>ProximitySensor</a> interface.
+<dl>
+  <dt><a>Mock Sensor Reading Values</a></dt>
+  <dd>
+    <pre class="idl">
+      dictionary ProximityReadingValues {
+        required double? distance;
+        required double? max;
+        required boolean? near;
+      };
+    </pre>
+  </dd>
+</dl>
+
+Extensions {#webdriver-extensions}
+=====================
+
+## Commands ## {#mock-sensor-commands}
+
+### Create mock sensor ### {#create-mock-sensor-command}
+
+<table>
+  <tbody>
+    <tr>
+      <th>HTTP Method</th>
+      <th><a lt="extension command URI template">URI Template</a></th>
+    </tr>
+    <tr>
+      <td>POST</td>
+      <td>/session/{session id}/sensor</td>
+    </tr>
+  </tbody>
+</table>
+
+The <dfn>create mock sensor</dfn> <a>extension command</a> simulates user
+creation of a <a>mock sensor</a>.
+
+The <a>remote end steps</a> are:
+1. Let |configuration| be the |configuration| argument, <a>converted to an IDL value</a>
+   of type {{MockSensorConfiguration}}. If this throws an exception, return a
+   <a>WebDriver error</a> with <a>WebDriver error code</a> <a>invalid argument</a>.
+2. Let |type| be the |configuration|.{{MockSensorConfiguration/mockSensorType}}, if the <a>current browsing context</a>
+   already has this |type| of [=mock sensor=], return a <a>WebDriver error</a> with <a>WebDriver error code</a>
+   [=mock sensor already created=].
+3. Run these sub-steps [=in parallel=] to create a [=mock sensor=] in <a>current browsing context</a>:
+  1. Set the [=mock sensor type|mock sensor's type=] to |type|.
+  2. Let |connected| be the |configuration|.{{MockSensorConfiguration/connected}},
+     force the result of invoking <a>connect to sensor</a> with associated <a>Sensor</a> instance as argument to |connected|.
+  3. If |configuration|.{{MockSensorConfiguration/maxSamplingFrequency}} is [=present=], then:
+    1. Set the [=mock sensor=]'s associated maximum supported sampling frequency to |configuration|.{{MockSensorConfiguration/maxSamplingFrequency}}.
+  4. If |configuration|.{{MockSensorConfiguration/minSamplingFrequency}} is [=present=], then:
+    1. Set the [=mock sensor=]'s minimum supported sampling frequency to |configuration|.{{MockSensorConfiguration/minSamplingFrequency}}.
+  5. Let all the |type| of <a>Sensor</a> objects' associated <a>platform sensor</a> be the |type|'s associated [=mock sensor=].
```

Comment:
I don't know what "Sensor objects'" refers to here. It seems like a `<var>` would be needed to refer to a concrete instance. A reference like `<a>Sensor</a>` describes the "Sensor" concept, not a tangible sensor that could influence an algorithm.